pkg/apis/apps/validation/validation.go

```diff
@@ -75,6 +75,48 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path) fi
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("podManagementPolicy"), spec.PodManagementPolicy, fmt.Sprintf("must be '%s' or '%s'", apps.OrderedReadyPodManagement, apps.ParallelPodManagement)))
 	}
 
+	switch spec.UpdateStrategy.Type {
+	case "":
+		allErrs = append(allErrs, field.Required(fldPath.Child("updateStrategy"), ""))
+	case apps.OnDeleteStatefulSetStrategyType, apps.RollingUpdateStatefulSetStrategyType:
+		if spec.UpdateStrategy.Partition != nil {
+			allErrs = append(
+				allErrs,
+				field.Invalid(
+					fldPath.Child("updateStrategy").Child("partition"),
+					spec.UpdateStrategy.Partition.Ordinal,
+					fmt.Sprintf("only allowed for updateStrategy '%s'", apps.PartitionStatefulSetStrategyType)))
+		}
+	case apps.PartitionStatefulSetStrategyType:
+		if spec.UpdateStrategy.Partition == nil {
```

Comment:
@kargakis, @smarterclayton  Do you feel strongly enough about this to gate LGTM? I'd be fine discussing this further offline and issuing PRs for removal in both the design and implementation after the fact, if that's where we land. If you think its a show stopper, I'd also be fine removing it in this PR provided we have general agreement. We can issue a PR to update the design concurrently.